src/Symfony/Component/TypeInfo/TypeResolver/StringTypeResolver.php

```diff
@@ -119,6 +120,40 @@ private function getTypeFromNode(TypeNode $node, ?TypeContext $typeContext): Typ
         }
 
         if ($node instanceof ConstTypeNode) {
+            if ($node->constExpr instanceof ConstFetchNode) {
+                $types = [];
+
+                $className = match (strtolower($node->constExpr->className)) {
+                    'self' => $typeContext->getDeclaringClass(),
+                    'static' => $typeContext->getCalledClass(),
+                    'parent' => $typeContext->getParentClass(),
+                    default => $node->constExpr->className,
+                };
+
+                foreach ((new \ReflectionClass($className))->getReflectionConstants() as $const) {
+                    if (preg_match('/^'.str_replace('\*', '.*', preg_quote($node->constExpr->name, '/')).'$/', $const->getName())) {
+                        $constValue = $const->getValue();
+
+                        $types[] = match (true) {
+                            true === $constValue,
+                            false === $constValue => Type::bool(),
+                            null === $constValue => Type::null(),
+                            \is_string($constValue) => Type::string(),
+                            \is_int($constValue) => Type::int(),
+                            \is_float($constValue) => Type::float(),
+                            \is_array($constValue) => Type::array(),
+                            $constValue instanceof \UnitEnum => Type::enum($constValue::class),
```

Comment:
We should have a default branch reporting a proper unsupported case (or using `mixed` as fallback) instead of letting PHP fail with a non-understandable ValueError due to a non-exhaustive match (non-understandable because the error message will talk about `true` not matching any branch, which gives no indication about `$constValue`)